### PR TITLE
Add menu to nteract-on-jupyter

### PR DESCRIPTION
**Pull request by theengineear**
🎉 first pass at some minimal menu functionality

I went with code-completeness > feature completeness. I.e., there's much more to flesh out here, but I tried my best to get this code right on the first pass so that we can easily extend.

TODO:

- [x] create `notebook-menu` presentation component (i made it a dir with`index.js` so it matches other components there)
- [x] connect presentation component to redux
- [x] snapshot test
- [x] test that we can mount and then assert that the appropriate actions are being called

### CODE REVIEW

**Comment by theengineear** [comment]
screen shot 2018-01-25 at 11 00 40 am

File: packages/core/src/components/notebook-menu/index.js [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+});
+
+const NotebookMenu = connect(mapStateToProps, mapDispatchToProps)(
+  PureNotebookMenu
+);
+
+// We export this for testing purposes.
+export { PureNotebookMenu };
+
+export default NotebookMenu;
```
Comment (rgbkrk) [inline packages/core/src/components/notebook-menu/index.js:212]: This is beautifully organized.

File: packages/core/src/components/notebook-menu/index.js [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+      </div>
+    );
+  }
+}
+
+// TODO: this forces the menu to re-render on cell focus. The better option is
+// to alter how the actions work to just target the currently focused cell.
+// That said, we *may* not have a great way of getting around this as we'll need
+// information about the current document to decide which menu items are
+// available...
```
Comment (rgbkrk) [inline packages/core/src/components/notebook-menu/index.js:185]: Yeah that's a tricky one. We probably need to work with the focused cell and if we need to create more actions (like the "run all").

**Review by rgbkrk — APPROVED** [review]

**Comment by rgbkrk** [comment]
I'm confused as to why that's failing on Appveyor, I'm going to go ahead and merge it, clear the cache on appveyor and figure out the build setup.

**Pull request merged**